flax/experimental/nnx/tests/compat/test_module.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2024 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+
+import jax
+import jax.numpy as jnp
+
+from flax.experimental import nnx
+from flax.experimental.nnx import compat
+
+
+class TestCompatModule:
+  def test_compact_basic(self):
+    class Linear(compat.Module):
+      dout: int
+
+      def setup(self):
+        self.count = 0
+
+      def __call__(self, x):
+        self.count += 1
+        if not hasattr(self, 'w'):
+          assert self.scope is not None
+          rngs = self.scope.rngs
+          self.w = nnx.Param(
+            jax.random.uniform(rngs(), (x.shape[-1], self.dout))
+          )
+          self.b = nnx.Param(jnp.zeros((self.dout,)))
+        return x @ self.w + self.b[None]
+
+    @dataclasses.dataclass
+    class Foo(nnx.compat.Module):
```

Comment:
```suggestion
    class Foo(compat.Module):
```
Change all `nnx.compat` references to `compat` for consistency?